app/src/androidTest/java/com/github/sdpcoachme/database/CachingStoreTest.kt

```diff
@@ -1217,9 +1221,24 @@ class CachingStoreTest {
         assertTrue(noError)
     }
 
+    @Test
+    fun getWeatherForecastRetrievesTwoWeeksForecast() {
 
-
-
+        val cachingStore = CachingStore(MockDatabase(),
+            ApplicationProvider.getApplicationContext<Context>().dataStoreTest,
+            ApplicationProvider.getApplicationContext()
+        )
+        val target = LatLng(LAUSANNE.latitude, LAUSANNE.longitude)
+        cachingStore.getWeatherForecast(target).thenApply {
+            runBlocking {
```

Comment:
Why do we need to wait like this in the callback of the future ?

Reply:
As `getWeatherForecast` always returns a `completedFuture`, we need to wait for the observableState to be actually updated.